src/appleseed/renderer/modeling/phasefunction/henyeyphasefunction.cpp

```diff
@@ -219,11 +219,12 @@ class HenyeyPhaseFunction
     virtual void evaluate_transmission(
         const ShadingRay&     volume_ray,
         const void*           data,
-        float                 distance,
+        const float           distance,
         Spectrum&             spectrum) const APPLESEED_OVERRIDE
     {
         extinction_coefficient(volume_ray, data, distance, spectrum);
-        spectrum = exp(-distance * spectrum);
+        for (int i = 0; i < spectrum.size(); ++i)
```

Comment:
Signed/unsigned mismatch, will not compile on Windows. Use `size_t`. In fact you should almost always use `size_t` (which I argue should be the default integer type in C++) unless you need signed values or a more compact memory representation.